accord-core/src/main/java/accord/utils/async/AsyncResult.java

```diff
@@ -79,7 +79,11 @@ default void setSuccess(V value)
         default void setFailure(Throwable throwable)
         {
             if (!tryFailure(throwable))
-                throw new IllegalStateException("Result has already been set on " + this);
+            {
+                IllegalStateException e = new IllegalStateException("Result has already been set on " + this);
```

Comment:
`ReadData.readComplete` some times throws causing this to happen.  Rather than dropping the throwable that we are rejecting, add it to the throwable we are throwing; that way logs can show what caused this to happen.

This doesn't happen once https://issues.apache.org/jira/browse/CASSANDRA-18364 is applied